embed.fnc

```diff
@@ -69,50 +69,58 @@
-:
-: The C flag is used instead for functions and their short names that need to
-:            be accessible everywhere, typically because they are called from a
-:            publicly available macro or inline function, but they are not for
-:            public use by themselves.  The documentation for these is placed
-:            in perlintern.  If no documentation exists, that fact is also
-:            noted in perlintern.
-:
-:            These really need the 'p' flag to avoid name space collisions.
-:
-:	     Some of these have been constructed so that the wrapper macro
-:	     names begin with an underscore to lessen the chances of a name
-:	     collision.  However, this is contrary to the C standard, and those
-:	     should be changed.
-:
-: The E flag is used instead for a function and its short name that is supposed
-:            to be used only in the core plus extensions compiled with the
-:            PERL_EXT symbol defined.  Again, on some platforms, the function
-:            will be visible everywhere, so one of the 'p' or 'S' flags is
-:            generally needed.  Also note that an XS writer can always cheat
-:            and pretend to be an extension by #defining PERL_EXT.
-:
-: The X flag is similar to the C flag in that the function (whose entry better
-:	     have the 'p' flag) is accessible everywhere on all platforms.
-:	     However the short name macro that normally gets generated is
-:	     suppressed outside the core.  (Except it is also visible in
-:	     PERL_EXT extensions if the E flag is also specified.)  This flag
-:	     is used for functions that are called from a public macro, the
-:	     name of which isn't derived from the function name.  You'll have
-:	     to write the macro yourself, and from within it, refer to the
-:	     function in its full 'Perl_' form with any necessary thread
-:	     context parameter.
+: care is taken to wrap it within something like #ifdef PERL_CORE..#endif or
+: more commonly PERL_IN_FILE_C (with FILE_C appropriately replaced).  Most (if
+: not all) C files define such a symbol before importing perl.h.  In general you
+: should restrict the exposure of your exports as much as possible, although
+: older code may not do so.  Be aware that non-static exports can be "over
+: exported" and things work out fine, but inline and static macros will cause
+: errors unless restricted to the specific file they are intended for, and the
+: generated PERL_ARGS_ macros will only be available to inline functions in the
+: appropriate context.  This why you will see sections listing each file in
+: turn, and is also why the file favours use of '#if defined()' form of
+: conditions instead of '#ifdef' as the latter only works with one symbol and
+: the former can be combined.  It is also why you will often see functions with
+: an 's' or 'i' export type grouped together into a single conditional block
+: separate from the functions from the same file with 'p' in them.
```

Comment:
I think this would be more readable in point form.

Reply:
Will do.

Reply:
I broke it out into paragraphs and made it breathe a bit. Check the latest push.